azext_iot/_params.py

```diff
@@ -809,7 +809,20 @@ def load_arguments(self, _):
 
     with self.argument_context("iot dps") as context:
         context.argument(
-            "dps_name", help="Name of the Azure IoT Hub device provisioning service"
+            "login",
+            options_list=["--login", "-l"],
+            validator=mode2_iot_login_handler,
+            help="This command supports an entity connection string with rights to perform action. "
+            'Use to avoid session login via "az login". '
+            "If both an entity connection string and name are provided the connection string takes priority. "
+            "Required if --dps-name is not provided.",
+            arg_group="Device Provisioning Service Identifier"
+        )
+        context.argument(
+            "dps_name",
+            options_list=["--dps-name", "-n"],
```

Comment:
Just to call out - one thing we have to be cognizant of, is if there is a chance `--dps-name` & `--hub-name` showing up in the same command causing issues because of the shared short alias `-n`. We have to ensure compatibility across repos (i.e. azure-cli repo which currently has the mgmt commands).

Reply:
Fortunately we should be able to deprecate arg options (like in CLI core).

Reply:
There isnt any command within the IoT Hub space that takes in dps_name and there isnt any command within the DPS space that takes in hub_name. For the command that links hubs in DPS, it currently takes in the hub connection string so there is no conflict.

Furthermore, if you look at `az iot dps access-policy create -h`:
![image](https://user-images.githubusercontent.com/73560279/141826738-7a94d871-e66c-484b-8359-42d46e5b1e92.png)
You can see that access-policy-name has the -n argument while the dps-name doesn't. The CLI core already handles duplicate param names by using the most specific scope. In fact if you look at the code, line 90 has the hub_name parameter with the -n argument.